src/softmax.jl

```diff
@@ -26,49 +32,20 @@ julia> softmax([1, 2, 3])
 
 See also [`logsoftmax`](@ref).
 """
-function softmax(xs::AbstractArray; dims=1)
-    max_ = maximum(xs, dims=dims)
-    exp_ = exp.(xs .- max_)
-    exp_ ./ sum(exp_, dims=dims)
+softmax(x; dims = 1) = softmax!(similar(x), x; dims)
```

Comment:
we still support julia < 1.5, so you will have to explicitly pass the keyword value

Reply:
also, this pattern will error on integer array inputs, that we still want to support, so we should promote to float types

Reply:
fixed.